### PR TITLE
[QUIC] added basic QUIC socket API

### PR DESCRIPTION
**Pull request by Mr3zee**
**Subsystem**
ktor-network

**Motivation**
These classes describe basic functionality around QUIC usage as transport protocol.  

**Solution**
`QUICServer` exposes `QUICStream` input and output channels. (`QUICStream` is a stub for now, as we plan to work around it at last inside QUIC itself).  `QUICServer` and `QUICSocketBase` are base classes, which will contain most of the QUIC logic we plan to implement.

### CODE REVIEW

File: ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/streams/QUICStreams.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    /**
+     * Receive a QUIC stream.
+     */
+    public suspend fun receive(): QUICStream = incoming.receive()
+}
+
+/**
+ * A channel for sending and receiving QUIC streams
+ */
+public interface QUICStreamReadWriteChannel : QUICStreamWriteChannel, QUICStreamReadChannel
```
Comment (e5l) [inline ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/streams/QUICStreams.kt:47]: Can be internal

**Review by e5l — APPROVED** [review]
lgtm for API

File: ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/streams/QUICStreams.kt (hunk 1 above)
Comment (Mr3zee) [reply ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/streams/QUICStreams.kt:47]: These interfaces are exposed via `aScoket().quic()`, do we want to hide it and allow only `HTTP/3` to use `QUIC`?

Comment (e5l) [reply ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/streams/QUICStreams.kt:47]: it's ok, let's keep it.

Feel free to merge after aprove

**Pull request merged**